column_buffer.go

```diff
@@ -2369,10 +2375,29 @@ func writeRowsFuncOfMap(t reflect.Type, schema *Schema, path columnPath) writeRo
 				if err := writeKeyValues(columns, empty, empty, levels); err != nil {
 					return err
 				}
-			} else {
-				elemLevels := levels
-				elemLevels.definitionLevel++
+				continue
+			}
+
+			elemLevels := levels
+			elemLevels.definitionLevel++
+
+			if compareKeys != nil {
```

Comment:
Since this simple condition is already present, do you think we could make the sort behavior optional?

Reply:
I'd rather wait to have a problem before increasing complexity by making this configurable, if sorting the keys is always better than not doing there's no need to maintain more features.